.github/workflows/check_license_header.yaml

```diff
@@ -0,0 +1,86 @@
+            echo "::error file=$filepath,line=1::check extension of $filepath";   \
+          done
+
+          test ! -s result-extension.check
+
+      - name: Check source code
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Compare first lines of each source file with reference in misc/HEADER
+          find src include test miniapp -type f                                   \
+               '('                                                                \
+                       -name '*.cpp'                                              \
+                   -o  -name '*.h'                                                \
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+               ')'                                                                \
+            | xargs -I{} sh -c                                                    \
+                "head -n9 {} | diff -sq - misc/HEADER > /dev/null || echo {}"     \
+            > result-cpp.check
+
+          # Generate an error message for each offending file
+          for filepath in `cat result-cpp.check`; do                              \
+            echo "::error file=$filepath,line=1::check the license in $filepath"; \
+          done
+
+          test ! -s result-cpp.check
+
+      - name: Check CMake and Python files
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Generate header license CMake compliant
+          sed 's|^//|#|g' misc/HEADER > HEADER_HASH
+
+          # Compare first lines of cmake files with the reference
+          find  '(' -type f ! -path './misc*' ')'                                 \
+                '('                                                               \
+                        -name 'CMakeLists.txt'                                    \
+                    -o  -name 'DLAF*.cmake'                                       \
+                    -o  -name 'DLAF*.cmake.in'                                    \
+                ')'                                                               \
+            | xargs -I{} sh -c                                                    \
+                "head -n9 {} | diff -sq - HEADER_HASH > /dev/null || echo {}"     \
+            >> result.check
+
+          # Look for the LICENSE inside the python file
+          function streambin { xxd -p $1 | tr -d "\n"; }
+          BIN_HEADER_CMAKE=`streambin HEADER_HASH`
+          for file in `find scripts -type f -name "*.py"`; do                     \
+            streambin $file | grep -sq $BIN_HEADER_CMAKE || echo $file;           \
+          done >> result.check
```

Comment:
I wasn't able to figure out an easier and more neat solution.

Since grep does not deal with multilines, the workaround is to see script files as "binary" and then look for the exact ASCII code sequence representing the LICENSE.